osu.Game/Screens/Play/HUD/KeysPerSecond/KeysPerSecondCalculator.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Timing;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Screens.Play.HUD.KeysPerSecond
+{
+    public class KeysPerSecondCalculator : Component
+    {
+        private readonly List<double> timestamps;
+
+        private InputListener? listener;
+
+        [Resolved]
+        private GameplayClock? gameplayClock { get; set; }
+
+        [Resolved(canBeNull: true)]
+        private DrawableRuleset? drawableRuleset { get; set; }
+
+        public InputListener Listener
+        {
+            set
+            {
+                onResetRequested?.Invoke();
+                listener = value;
+                listener.OnNewInput += addTimestamp;
+            }
+        }
+
+        private event Action? onResetRequested;
+
+        private IClock? workingClock => drawableRuleset?.FrameStableClock;
+
+        private double baseRate;
+
+        private double rate
+        {
+            get
+            {
+                if (gameplayClock != null)
+                {
+                    if (gameplayClock.TrueGameplayRate > 0)
```

Comment:
Please merge these conditions into `gameplayClock?.TrueGameplayRate > 0`. The condition will evaluate to false for null click.

Reply:
Fixed in 9dc8065